NDTensors/src/lib/BlockSparseArrays/src/blocksparsearray/blockdiagonalarray.jl

```diff
@@ -0,0 +1,60 @@
+# type alias for block-diagonal
+using LinearAlgebra: Diagonal
+
+const BlockDiagonal{T,A,Axes,V<:AbstractVector{A}} = BlockSparseMatrix{
+  T,A,Diagonal{A,V},Axes
+}
+
+function BlockDiagonal(blocks::AbstractVector{<:AbstractMatrix})
+  return BlockSparseArray(
+    Diagonal(blocks), (blockedrange(size.(blocks, 1)), blockedrange(size.(blocks, 2)))
+  )
+end
+
+# Cast to block-diagonal implementation if permuted-blockdiagonal
+function find_permute_blockdiagonal(A)
+  @show keys = map(Base.Fix2(getproperty, :n), collect(block_stored_indices(A)))
```

Comment:
I think this is the canonical way to convert `Block(1, 1)` to `(1, 1)`, as an alternative to using field access:
```julia
Int.(Tuple(Block(1, 1)))
```

Reply:
I played around with this, but within the current constraints I don't think I can make that work. I really need a vector of blocks in this case, in other words linear slicing into the block sparse matrix. `A[vec_of_blockinds]` just cannot work, for the reasons we discussed previously, because the slicing wants to return a `BlockArray`, which would then be a vector of matrices that all have different sizes.

My way around this is to perform linear index slicing into the `blocks`, which I think is fair to assume to return a vector of blocks (not wrapped into a `BlockArray`). I agree that it would be nicer to not have to do this, but I would also argue that it is not outside the allowed operations of the interface. `blocks` should return an `AbstractArray` containing the different blocks, and I should be allowed to index into that to obtain different blocks.

Reply:
Hmm I'm not quite following, let's discuss this.